### DISCUSSION TITLE
How to use cogent3 to perform pairwise alignment?

### DISCUSSION THREAD

**Discussion by xingjianleng** [cogent3/cogent3#1015] [Q&A]
Hi!

I'm using cogent3 to align two sequences from fasta. But using the code below it says that two sequences don't have the same length and throws an error.

```
from cogent3 import Alignment
from cogent3 import load_unaligned_seqs
from pathlib import Path

path = Path("~/repos/corona-subset.fasta").expanduser().absolute()
seqs = load_unaligned_seqs(path, moltype='dna')
aln = Alignment(seqs)
# aln.alignment_quality()
```

I'm wondering how to align sequences with different length? Also, how to specify which alignment algorithm to use (e.g. Needleman Wunsch, partial order graph)?

Many thanks!

**Comment by GavinHuttley** [comment]
The `Alignment` object represents an existing alignment. You want to create the alignment.

```python
from cogent3.align.align import global_pairwise, make_dna_scoring_dict

s = make_dna_scoring_dict(10, -1, -8)

aln, score = global_pairwise(*seqs.seqs, s, 10, 2, return_score=True)
```

(I'm assuming `seqs` has two sequences.)
